### PR TITLE
Avoid closing socket too soon

### PR DESCRIPTION
**Pull request by luisfpinto**

### CODE REVIEW

File: lib/plug.js [hunk 1]
```diff
@@ -202,7 +202,7 @@ class Plug {
 
     socket.on('connect', () => {
       socket.write(encryptWithHeader(payload), () => {
-        socket.end()
+        // socket.end() Don't close the socket when you send data, do it when you get a response or error.
```
Comment (jsdario) [inline lib/plug.js:205]: Tiene mucho sentido, solo tendría que ver si es posible mandar info sin esperar respuesta, en ese caso habría quizás que cerrarlo

Comment (luisfpinto) [reply lib/plug.js:205]: Cierto es, si en el lado cliente no hace falta ningún tipo de respuesta del socket se puede seguir cerrando ahí!

**Comment by jsdario** [comment]
[When you execute a charm, the state of the TpLink switches isn't updated in the home view but also, charms executed that contains a TpLink are always marked as charm failed in the analytics even it has a success](https://trello.com/c/hRmhHs5c/173-when-you-execute-a-charm-the-state-of-the-tplink-switches-isnt-updated-in-the-home-view-but-also-charms-executed-that-contains-a)

**Review by jsdario — APPROVED** [review]
All methods eventually call `get`, that awaits for an answer.

**Pull request merged**